### DISCUSSION TITLE
Schism Tracker Accessibility Edition

### DISCUSSION THREAD

**Discussion by a11cf0** [schismtracker/schismtracker#558] [Show and tell]
Hello.
Here's a new interesting thing to try for all blind/visually impaired tracking enthusiasts out there.
I made an experimental fork of Schism Tracker with a self-voicing accessibility mode, that works on Windows and Linux. It may possibly work on macOS too, but this has never been tested.
It can speak through NVDA, JAWS, Narrator or SAPI5 on Windows, Speech Dispatcher on Linux/*BSD ant theoretically AVSpeach on macOS.
Since this is my first try at programming in C, the accessibility implementation is certainly far from optimal but should already be mostly usable.
So, it's now possible to get an Impulse Tracker/Scream Tracker style experience in an accessible way, if anyone has ever wanted this. Also this seems to be the first tracker with cross-platform accessibility, as far as I know.
Further information about this project is available at this link. https://github.com/a11cf0/schismtracker-a11y/blob/master/docs/accessibility.md
There are currently no binary releases, but working Windows builds are already available as artifacts.
Any feedback is off course appreciated.
Cheers.

**Comment by sagamusix** [comment]
As I already said on the OpenMPT forums, I think it would be great if this could be cleaned up and integrated into mainline Schism Tracker, if only as an optional feature (the dependency on a C++ library might be a showstopper for some people).

**Comment by mrpapersonic** [comment]
This is super cool!

As Saga Musix said, this ought to be an optional feature. It would also be nice if it could be dynamically loaded and enabled at runtime so e.g. Linux users won't need the extra library and Windows users can remove it if they wish. Examples of this in the current codebase can be found in `fmt/win32mf.c`, `fmt/flac.c`, `sys/alsa/midi-alsa.c`, `sys/jack/midi-jack.c`, and everything under `sys/sdl12` and `sys/sdl2` for a reference.
